### PR TITLE
MINOR: Rename InitPidRequest/InitPidResponse to InitProducerIdRequest/InitProducerIdResponse

### PR DESCRIPTION
**Pull request by hachikuji**

### CODE REVIEW

**Comment by guozhangwang** [comment]
LGTM!

**Comment by ijuma** [comment]
Thanks for cleaning this up.

**Comment by hachikuji** [comment]
@guozhangwang Unless you say otherwise, I'll hold off merging this until your TC patch is in.

**Comment by hachikuji** [comment]
Merging to trunk.

File: clients/src/main/java/org/apache/kafka/common/protocol/Errors.java [hunk 1]
```diff
@@ -439,7 +439,8 @@ public ApiException build(String message) {
                 return new InvalidTxnStateException(message);
             }
         }),
-    INVALID_PID_MAPPING(49, "The PID mapping is invalid",
+    INVALID_PRODUCER_ID_MAPPING(49, "The producer attempted to use a producerId which is not currently assigned to " +
+            "its transactionalId",
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/protocol/Errors.java:443]: Super nit: do we really want to say `producerId` and `transactionalId` in the message? Or should it simply be `producer id` and `transactional id`?

File: core/src/main/scala/kafka/utils/ZkUtils.scala [hunk 2]
```diff
@@ -64,7 +64,7 @@ object ZkUtils {
   val BrokerSequenceIdPath = s"$BrokersPath/seqid"
   val ConfigChangesPath = s"$ConfigPath/changes"
   val ConfigUsersPath = s"$ConfigPath/users"
-  val PidBlockPath = "/latest_pid_block"
+  val ProducerIdBlockPath = "/latest_pid_block"
```
Comment (ijuma) [inline core/src/main/scala/kafka/utils/ZkUtils.scala:67]: Do we want to update this to be "latest_producer_id_block"?

File: clients/src/main/java/org/apache/kafka/common/protocol/Errors.java (hunk 1 above)
Comment (hachikuji) [reply clients/src/main/java/org/apache/kafka/common/protocol/Errors.java:443]: Sure, I can change this in #3040.

File: core/src/main/scala/kafka/utils/ZkUtils.scala (hunk 2 above)
Comment (hachikuji) [reply core/src/main/scala/kafka/utils/ZkUtils.scala:67]: I debated it. The name seemed kind of long, but maybe we should just do it.

Comment (ijuma) [reply core/src/main/scala/kafka/utils/ZkUtils.scala:67]: Here's a PR that does this and a few other renames: https://github.com/apache/kafka/pull/3041

**Pull request closed without merging**